### PR TITLE
fix: align pickup times with API data

### PR DESCRIPTION
**Pull request by MPClerise**
Fixed weekday ordering and mapped pickup times accurately to match API data.

INT-663

### CODE REVIEW

File: apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue [hunk 1]
```diff
@@ -1,66 +1,94 @@
...
+  const hoursArray = allWeekdays.map((weekday) => {
+    const openingHour = (pickupLocation.value?.openingHours ?? []).find(({ weekday: wd }) => wd === weekday);
+    const date = createNextDate(weekday);
+    const formattedDay = useDateFormat(date);
+
+    const time: StartEndDate | undefined = openingHour && Array.isArray(openingHour.hours) && openingHour.hours.length > 0 ? openingHour.hours[0] : undefined;
+
+    let timerange = 'CLOSED';
+    if (time) {
+      const start = time.start.date.substring(11, 16);
```
Comment (FlorianSDV) [inline apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue:71]: Is there a way to get this data without using `substring()`? I am worried that this might not work in the future if the formatting of date changes.

**Review by FlorianSDV — CHANGES_REQUESTED** [review]

File: libs/shared/src/utils/createTimestamp.ts [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 import {type Timestamp} from '@myparcel/sdk';
-import {DEFAULT_TIMEZONE, DEFAULT_TIMEZONE_TYPE, API_DATE_FORMAT} from '../data';
+import {API_DATE_FORMAT} from '../data';
 
 export const createTimestamp = (date: DateLike = new Date()): Timestamp => {
   return {
     date: format(normalizeDate(date), API_DATE_FORMAT),
-    timezone: DEFAULT_TIMEZONE,
-    timezone_type: DEFAULT_TIMEZONE_TYPE,
+    timezone: 'GMT',
```
Comment (joerivanveen) [inline libs/shared/src/utils/createTimestamp.ts:9]: I think if you leave timezone and timezone_type to the defaults, it does not influence your pr?

File: apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue (hunk 1 above)
Comment (joerivanveen) [reply apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue:71]: I agree this can be clearer (no magic numbers), but the string is a fixed ISO string that will likely not change over the lifetime of the application.

File: apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue [hunk 3]
```diff
@@ -1,66 +1,94 @@
...
-        weekday: capitalize(isTodayOrTomorrow ? formattedDay.relative.value : formattedDay.weekday.value),
-        timeString,
-      };
-    })
-    .sort((a, b) => a.date.getTime() - b.date.getTime());
+  const allWeekdays = [DAY_SUNDAY, DAY_MONDAY, DAY_TUESDAY, DAY_WEDNESDAY, DAY_THURSDAY, DAY_FRIDAY, DAY_SATURDAY];
+
+  const hoursArray = allWeekdays.map((weekday) => {
+    const openingHour = (pickupLocation.value?.openingHours ?? []).find(({ weekday: wd }) => wd === weekday);
+    const date = createNextDate(weekday);
```
Comment (joerivanveen) [inline apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue:64]: How come this is not moving the days up like what we noticed earlier? Did we not find out you don’t need to supply the date?

File: apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue [hunk 4]
```diff
@@ -1,66 +1,94 @@
...
-        date,
-        weekday: capitalize(isTodayOrTomorrow ? formattedDay.relative.value : formattedDay.weekday.value),
-        timeString,
-      };
-    })
-    .sort((a, b) => a.date.getTime() - b.date.getTime());
+  const allWeekdays = [DAY_SUNDAY, DAY_MONDAY, DAY_TUESDAY, DAY_WEDNESDAY, DAY_THURSDAY, DAY_FRIDAY, DAY_SATURDAY];
+
+  const hoursArray = allWeekdays.map((weekday) => {
+    const openingHour = (pickupLocation.value?.openingHours ?? []).find(({ weekday: wd }) => wd === weekday);
```
Comment (joerivanveen) [inline apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue:63]: This is complicated for me, but also a bit confusing: I think ‘openingHour’ is not the correct variable name, since it contains an array?

File: apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue [hunk 5]
```diff
@@ -1,66 +1,94 @@
...
+    let timerange = 'CLOSED';
+    if (time) {
+      const start = time.start.date.substring(11, 16);
+      const end = time.end.date.substring(11, 16);
+      timerange = `${start} - ${end}`;
+    }
+
+    return {
+      date: formattedDay,
+      weekday: weekDaysMap[weekday as keyof typeof weekDaysMap],
```
Comment (joerivanveen) [inline apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue:78]: Will this be translated to the locale?

File: apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue [hunk 6]
```diff
@@ -1,66 +1,94 @@
...
+
+    let timerange = 'CLOSED';
+    if (time) {
+      const start = time.start.date.substring(11, 16);
+      const end = time.end.date.substring(11, 16);
+      timerange = `${start} - ${end}`;
+    }
+
+    return {
+      date: formattedDay,
```
Comment (joerivanveen) [inline apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue:77]: Is this used further down, ie does it matter what date you supply?

**Review by joerivanveen — CHANGES_REQUESTED** [review]

File: libs/shared/src/utils/createTimestamp.ts (hunk 2 above)
Comment (MPClerise) [reply libs/shared/src/utils/createTimestamp.ts:9]: Updated the createTimestamp function to remove unused timezone and timezone_type fields as they are not referenced or required elsewhere. Simplified the implementation to focus only on the date field while ensuring functionality remains intact. Tests confirm the change works as expected.

Comment (MPClerise) [reply libs/shared/src/utils/createTimestamp.ts:9]: I reset these to their defaults, good suggestion thank you

**Review by RichardPerdaan — APPROVED** [review]

File: apps/delivery-options/src/composables/usePickupLocation.ts [hunk 7]
```diff
@@ -11,9 +11,18 @@ interface UsePickupLocation {
...
+    () => get(locationCode),
+    (newCode) => {
+      currentLocationCode.value = newCode;
+    },
+    {immediate: true},
+  );
 
   const pickupLocation = computed(() => {
-    return toValue(locations).find((location) => location.locationCode === get(locationCode));
+    return toValue(locations).find((location) => location.locationCode === currentLocationCode.value);
```
Comment (FreekVR) [inline apps/delivery-options/src/composables/usePickupLocation.ts:25]: `toValue` and `get` are mostly the same, and are now used interchangeably here, along with direct `.value` access.

Prefer to use `toValue` where possible, as that's in Vue core.

File: apps/delivery-options/src/composables/usePickupLocation.ts [hunk 8]
```diff
@@ -11,9 +11,18 @@ interface UsePickupLocation {
 
 export const usePickupLocation = (locationCode: MaybeRef<string | undefined>): UsePickupLocation => {
   const {locations} = useResolvedPickupLocations();
+  const currentLocationCode = ref(get(locationCode));
```
Comment (FreekVR) [inline apps/delivery-options/src/composables/usePickupLocation.ts:14]: Perhaps I don't completely understand, but as I read this, this is just setting up a local ref that will simply update with the existing `locationCode` value whenever it changes. What exactly is this change intending to do here and why wasn't the original `locationCode` working for this purpose?

Would be good to add in an inline code comment as well.

File: apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue [hunk 9]
```diff
@@ -5,69 +5,57 @@
...
-    .sort((a, b) => a.date.getTime() - b.date.getTime());
-});
-
-const filteredOpeningHours = computed(() => {
-  return resolvedShowAll.value ? openingHours.value : openingHours.value.slice(0, SHOWN_OPENING_HOURS);
-});
-
-onDeactivated(() => {
-  mutableShowAll.value = false;
+  const openingHoursData = pickupLocation.value?.openingHours ?? [];
```
Comment (FreekVR) [inline apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue:38]: prefer using `toValue` here, too

File: apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue [hunk 10]
```diff
@@ -5,69 +5,57 @@
...
+
+  return weekDaysMap.map((day, index) => {
+    const dayData = openingHoursData.find((entry: {weekday: number}) => entry.weekday === index);
+
+    if (dayData?.hours && dayData.hours.length > 0) {
+      const times = dayData.hours[0];
+
+      if (times.start && times.end) {
+        const timeString = `${times.start.date.substring(11, 16)} - ${times.end.date.substring(11, 16)}`;
```
Comment (FreekVR) [inline apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue:47]: I don't like having random numbers in code, as its hard to see what they signify. Prefer wrapping these in named constants so it's immediately clear what they're intending to do. This code currently seems pretty error-prone, but it's hard to say without knowing how consistent the `date` input string is supposed to be -- prefer to work with dates using `Date` or `date-fns` rather than string data.

**Review by FreekVR — CHANGES_REQUESTED** [review]

File: apps/delivery-options/src/composables/usePickupLocation.ts (hunk 8 above)
Comment (CleriseSwart) [reply apps/delivery-options/src/composables/usePickupLocation.ts:14]: The change was intended to keep track of updates to locationCode reactively. However, after reviewing, I realized the original approach using locationCode directly with toValue already achieves this effectively. The ref and watch logic introduced redundancy, so I removed it to streamline the code and maintain simplicity.

Comment (MPClerise) [reply apps/delivery-options/src/composables/usePickupLocation.ts]: The change was intended to keep track of updates to locationCode reactively. However, after reviewing, I realized the original approach using locationCode directly with toValue already achieves this effectively. The ref and watch logic introduced redundancy, so I removed it to streamline the code and maintain simplicity.

File: apps/delivery-options/src/composables/usePickupLocation.ts [hunk 11]
```diff
@@ -12,8 +11,9 @@ interface UsePickupLocation {
 export const usePickupLocation = (locationCode: MaybeRef<string | undefined>): UsePickupLocation => {
   const {locations} = useResolvedPickupLocations();
 
+  // Use toValue to consistently extract the current value of locationCode
```
Comment (EdieLemoine) [inline apps/delivery-options/src/composables/usePickupLocation.ts:14]: This comment is not necessary, we do this everywhere

File: .idea/jsLinters/eslint.xml
Comment (EdieLemoine) [inline .idea/jsLinters/eslint.xml:1]: Please revert this

File: apps/delivery-options/src/composables/useLanguage.ts [hunk 12]
```diff
@@ -16,7 +16,8 @@ interface UseLanguage {
 
   setLocale(locale: string): void;
   setStrings(strings: Record<string, string>): void;
-  translate(translatable: AnyTranslatable): string;
+
+  translate(translatable: string): string;
```
Comment (EdieLemoine) [inline apps/delivery-options/src/composables/useLanguage.ts:20]: Why did you change this? Now there are type errors in places where this is called with a `Translatable`

File: apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue [hunk 13]
```diff
@@ -5,69 +5,60 @@
       v-text="translate(OPENING_HOURS)" />
 
     <div
-      v-for="hours in filteredOpeningHours"
+      v-if="openingHours.length === 0"
+      class="mp-content-between"
+      v-text="translate(CLOSED)"></div>
```
Comment (EdieLemoine) [inline apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue:10]: ESLint should make you use an inline closing tag here. Please re-enable ESLint and lint all the files you've changed. 

IntellIJ IDEs sometimes give an error when trying to lint .vue files, to "fix" it for a bit, restart the ESLint service.

File: apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue [hunk 14]
```diff
@@ -5,69 +5,60 @@
...
-      const formattedDay = useDateFormat(date);
+  const openingHoursData = toValue(pickupLocation)?.openingHours ?? [];
 
-      const isTodayOrTomorrow = isToday(date) || isSameDay(addDays(createUtcDate(), 1), date);
+  return weekDaysMap.map((day, index) => {
+    const dayData = openingHoursData.find((entry: {weekday: number}) => entry.weekday === index);
 
-      const time: StartEndDate = hours?.[0];
-      const timeString = time ? useTimeRange(time.start.date, time.end.date).value : translate(CLOSED);
+    if (dayData?.hours && dayData.hours.length > 0) {
```
Comment (EdieLemoine) [inline apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue:44]: Try to avoid using nested blocks, they're badly readable. Also, with optional chaining you don't need multiple if statements.

Example of how to improve this: (untested)

```ts
  return weekDaysMap.map((day, index) => {
    const dayData = openingHoursData.find((entry: {weekday: number}) => entry.weekday === index);
    const times = dayData?.hours?.[0];

    if (!(times?.start && times?.end)) {
      return {
        weekday: day,
        timeString: CLOSED,
      };
    }

    const startTime = format(new Date(times.start.date), 'HH:mm');
    const endTime = format(new Date(times.end.date), 'HH:mm');
    const timeString = `${startTime} – ${endTime}`;

    return {
      weekday: day,
      timeString,
    };
  });
```

Tip: IntellIJ has a "Flip if-else statement" intention (option+enter) that makes this very easy.

File: apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue [hunk 15]
```diff
@@ -5,69 +5,60 @@
...
 
 const {translate} = useLanguage();
+const locationCodeRef = toRef(props, 'locationCode');
+const {pickupLocation} = usePickupLocation(locationCodeRef);
 
-const {pickupLocation} = usePickupLocation(propRefs.locationCode);
-
-const mutableShowAll = ref(false);
-const resolvedShowAll = computed(() => props.expanded || mutableShowAll.value);
+const weekDaysMap = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];
```
Comment (EdieLemoine) [inline apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue:36]: This doesn't translate to other languages anymore. Use the `useWeekdays` composable.

File: apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue [hunk 16]
```diff
@@ -5,69 +5,60 @@
...
-      };
-    })
-    .sort((a, b) => a.date.getTime() - b.date.getTime());
-});
-
-const filteredOpeningHours = computed(() => {
-  return resolvedShowAll.value ? openingHours.value : openingHours.value.slice(0, SHOWN_OPENING_HOURS);
-});
+      if (times.start && times.end) {
+        const startTime = format(new Date(times.start.date), 'HH:mm');
```
Comment (EdieLemoine) [inline apps/delivery-options/src/views/MyParcelDeliveryOptions/Pickup/PickupLocationOpeningHours/PickupLocationOpeningHours.vue:48]: there's an useDateFormat composable, can't it be used for this?

File: libs/shared/src/utils/createTimestamp.ts [hunk 17]
```diff
@@ -1,9 +1,10 @@
 import {format} from 'date-fns';
 import {type DateLike, normalizeDate} from '@vueuse/core';
-import {type Timestamp} from '@myparcel/sdk';
-import {DEFAULT_TIMEZONE, DEFAULT_TIMEZONE_TYPE, API_DATE_FORMAT} from '../data';
+import {API_DATE_FORMAT, DEFAULT_TIMEZONE_TYPE, DEFAULT_TIMEZONE} from '../data';
 
-export const createTimestamp = (date: DateLike = new Date()): Timestamp => {
+export const createTimestamp = (
+  date: DateLike = new Date(),
+): {date: string; timezone: string; timezone_type: number} => {
```
Comment (EdieLemoine) [inline libs/shared/src/utils/createTimestamp.ts:7]: Why was this changed? It seems to work fine with the imported type

**Review by EdieLemoine — CHANGES_REQUESTED** [review]
This is not a feature, it's a bug fix, so the commit type should be `fix`. Also, the "Resolves #..." thing is for GitHub issues, when referring to a Jira ticket just put `INT-xxx` in the body. This is also mentioned in the "GitHub pull request workflow" on the wiki.

**Review by joerivanveen — APPROVED** [review]
